example/multi_fem/TestCuda.cpp

```diff
@@ -86,12 +86,12 @@ void test_cuda_fixture( comm::Machine machine ,
     dev_count && dev_count <= comm_size ? comm_rank % dev_count : 0 ;
   const size_t gang_count = 0 ;
 
-  Kokkos::HostSpace::execution_space::initialize();
-  Kokkos::Cuda::SelectDevice select_device( dev_rank );
-  Kokkos::Cuda::initialize( select_device );
+  Kokkos::InitArguments args;
+  args.device_id = dev_rank;
+  Kokkos::initialize(args);
```

Comment:
Prefer `ScopeGuards`?

Reply:
Possibly... could be a future improvement. For now, it is clear enough that all the work is inside one function call between the two, so there should be no lifetime issues.